modules/run-vault/run-vault

```diff
@@ -103,14 +110,39 @@ function generate_vault_config {
   local readonly cluster_port="$4"
   local readonly config_dir="$5"
   local readonly user="$6"
+  local readonly enable_s3_backend="$7"
+  local readonly s3_bucket="$8"
+  local readonly s3_bucket_region="$9"
   local readonly config_path="$config_dir/$VAULT_CONFIG_FILE"
 
   local instance_ip_address
   instance_ip_address=$(get_instance_ip_address)
 
   log_info "Creating default Vault config file in $config_path"
-  cat > "$config_path" <<EOF
-storage "consul" {
+  listener_config=$(cat <<EOF
```

Comment:
We follow the [Google Bash Style Guide](https://google.github.io/styleguide/shell.xml), so please declare the new variables in this PR using `local` and, where possible, `readonly`. 

E.g.,:

```bash
local readonly consul_storage_type="storage"
local s3_config=""

if [[ "$enable_s3_backend" == "true" ]]; then
  ...
fi

local readonly listener_config=$(cat <<EOF
...
EOF
)

local readonly consul_storage_config=$(cat <<EOF
...
EOF
)
```